hvpy/core.py

```diff
@@ -0,0 +1,22 @@
+import requests
+from hvpy.io import HvpyParameters, OutputType
+
+
+def parse_response(response: requests.Response, output_parameters: OutputType):
+
+    if output_parameters == OutputType.Raw:
+        return response.content
+    elif output_parameters == OutputType.String:
+        return str(response.content)
+    elif output_parameters == OutputType.Json:
+        return response.json()
+
+
+def execute_api_call(url: str, input_parameters: HvpyParameters, output_parameters: OutputType):
+
+    response = requests.get(url, params=input_parameters)
+    # check if we have a valid response
+    if response.status_code != 200:
```

Comment:
There is if I recall, a raise_for_status method on the response. That might be neater, but you will have to check.

Reply:
+1 raise_for_status would be good here.